test/StableJoeStaking.test.js

```diff
@@ -0,0 +1,692 @@
+
+      // Making sure that `pendingReward` still return the accurate tokens even after updating pools
+      await this.sJoe.updateReward(this.rewardToken.address);
+      expect(
+        await this.sJoe.pendingReward(
+          this.alice.address,
+          this.rewardToken.address
+        )
+      ).to.be.equal(ethers.utils.parseEther("1"));
+
+      await this.rewardToken
+        .connect(this.joeMaker)
+        .transfer(this.sJoe.address, ethers.utils.parseEther("1"));
+      await increase(86_400);
+
+      // Should be equal to 2, the previous reward and the new one
+      expect(
+        await this.sJoe.pendingReward(
+          this.alice.address,
+          this.rewardToken.address
+        )
+      ).to.be.equal(ethers.utils.parseEther("2"));
+
+      // Making sure that `pendingReward` still return the accurate tokens even after updating pools
+      await this.sJoe.updateReward(this.rewardToken.address);
+      expect(
+        await this.sJoe.pendingReward(
+          this.alice.address,
+          this.rewardToken.address
+        )
+      ).to.be.equal(ethers.utils.parseEther("2"));
+    });
+
+    it("should allow deposits and withdraws of multiple users and distribute rewards accordingly", async function () {
+      await this.sJoe
+        .connect(this.alice)
+        .deposit(ethers.utils.parseEther("100"));
+      expect(await this.joe.balanceOf(this.alice.address)).to.be.equal(
+        ethers.utils.parseEther("900")
+      );
+      expect(await this.joe.balanceOf(this.sJoe.address)).to.be.equal(
+        ethers.utils.parseEther("97")
+      );
+      expect(
+        (await this.sJoe.getUserInfo(this.alice.address, this.joe.address))[0]
+      ).to.be.equal(ethers.utils.parseEther("97"));
+
+      await this.sJoe.connect(this.bob).deposit(ethers.utils.parseEther("200"));
+      expect(await this.joe.balanceOf(this.bob.address)).to.be.equal(
+        ethers.utils.parseEther("800")
+      );
+      expect(await this.joe.balanceOf(this.sJoe.address)).to.be.equal(
+        ethers.utils.parseEther("291")
+      );
+      expect(
+        (await this.sJoe.getUserInfo(this.bob.address, this.joe.address))[0]
+      ).to.be.equal(ethers.utils.parseEther("194"));
+
+      await this.sJoe
+        .connect(this.carol)
+        .deposit(ethers.utils.parseEther("300"));
+      expect(await this.joe.balanceOf(this.carol.address)).to.be.equal(
+        ethers.utils.parseEther("700")
+      );
+      expect(await this.joe.balanceOf(this.sJoe.address)).to.be.equal(
+        ethers.utils.parseEther("582")
+      );
+      expect(
+        (await this.sJoe.getUserInfo(this.carol.address, this.joe.address))[0]
+      ).to.be.equal(ethers.utils.parseEther("291"));
+
+      await this.rewardToken
+        .connect(this.joeMaker)
+        .transfer(this.sJoe.address, ethers.utils.parseEther("6"));
+      await this.sJoe.updateReward(this.rewardToken.address);
+      await increase(86_400);
```

Comment:
`86400` (either type all numbers with underscore, or all numbers with, but not some with and some without).